python/applications/ranking/data/scripts/create_dataset.py

```diff
@@ -0,0 +1,148 @@
+    rows_df = []
+    seed_dict = {}  # the keys are source data columns, the values are lists of source data column values
+    seed_dict_labeled_pos = {}  # option to get characteristics of labeled results separately
+    seed_dict_labeled_neg = {}
+
+    name_num_sequences = [x['serving_info']['name'] for x in feature_config.all_features if x['name']=='num_sequences'][0]
+    name_label_rank = [x['serving_info']['name'] for x in feature_config.all_features if x['name']=='label_rank'][0]
+    name_highval_feature = [x['serving_info']['name'] for x in feature_config.all_features if x['name']=='high_val_feature_1'][0]
+
+    for mycol in list(example_data.columns):
+        # Generate data distributions from sample data
+        seed_dict[mycol] = list(example_data[mycol].values)
+        seed_dict_labeled_pos[mycol] = list(example_data[example_data[feature_config.get_label('name')] == 1].values)
+        seed_dict_labeled_neg[mycol] = list(example_data[example_data[feature_config.get_label('name')] == 0].values)
+
+    seed_dict[name_num_sequences] = list(filter(lambda x: x <= max_num_records, seed_dict[name_num_sequences]))
+
+    for _ in range(num_samples):
+        # Generate a synthetic query ID
+        seed_id = str(random.sample(seed_dict[feature_config.get_query_key('name')], 1)[0])
+        q_id = ''.join(random.sample(seed_id, len(seed_id)))
+
+        # Sample the number of results and label rank from example data
+        q_nseq = random.sample(seed_dict[name_num_sequences], 1)[0]
+        q_labelrank = random.sample(seed_dict[name_label_rank], 1)[0]
+
+        for i in range(int(q_nseq)):
+            is_labeled_pos = (i + 1 == q_labelrank)
+            row_dict = {
+                feature_config.get_query_key('name'): q_id,
+                name_num_sequences: q_nseq,
+                name_label_rank: q_labelrank,
+                feature_config.get_rank('name'): i + 1
+            }
+            for mycol in list(example_data.columns):
+                # Could also let user specify distribution to sample from for each feature
+                if mycol not in row_dict.keys() and is_labeled_pos:
+                    # row_dict[mycol] = random.sample(seed_dict_labeled_pos[mycol])
+                    row_dict[mycol] = None  # Default value, check type # Could also be functions we pass in
+                elif mycol not in row_dict.keys() and not is_labeled_pos:
+                    # row_dict[mycol] = random.sample(seed_dict_labeled_neg[mycol])
+                    row_dict[mycol] = None
+
+            # Create catastrophic failure # Rename, possible to extenuate this for each high value feature
+            # {high_value_feature_1:[label1_val, label2_val], high_value_feature_2: [unclicked_val, clicked_val]}
+            # Not creating examples of non-catastrophic failure, ie where the clicked record is not a name match
+            # Configurable feature to pass in here! Pass in: feature name
```

Comment:
@calliebradley What is `highval_feature`? I think we need a better name.

Reply:
@lastmansleeping This was Jake's suggestion for cat failure. If you don't like cat failure or highval feature, feel free to suggest something else ;) Otherwise I think this is enough for now. The 'name_match' makes it more explicit.